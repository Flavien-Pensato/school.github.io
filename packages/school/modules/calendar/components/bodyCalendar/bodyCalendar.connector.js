import { connect } from 'react-redux';

import { BodyCalendar } from './bodyCalendar.component';

import { getCurrentWeek } from '../../calendar.selectors';
import { fetchWeekAction } from '../../calendar.actions';

const mapStateToProps = (state, ownProps) => ({
  week: getCurrentWeek(state, ownProps.selectedWeek),
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  fetchWeek: () => dispatch(fetchWeekAction(ownProps.selectedWeek)),
});

export const BodyCalendarConnected = connect(
  mapStateToProps,
  mapDispatchToProps,
)(BodyCalendar);
