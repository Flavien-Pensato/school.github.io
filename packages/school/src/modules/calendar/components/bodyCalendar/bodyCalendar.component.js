import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

export class BodyCalendar extends Component {
  componentDidMount() {
    const { fetchWeeksAction, selectedWeek } = this.props;

    fetchWeeksAction(selectedWeek);
  }

  componentWillReceiveProps(nextProps) {
    const { fetchWeeksAction, selectedWeek } = nextProps;


    if (selectedWeek !== this.props.selectedWeek) {
      fetchWeeksAction(selectedWeek);
    }
  }


  render() {
    const { week } = this.props;

    if (_.isEmpty(week)) {
      return (
        <span>Pas de semaine programmée</span>
      );
    }

    return (
      <div id="planning">
        <table className="f7 w-100 center" cellSpacing="0">
          <thead>
            <tr>
              <th className="fw6 bb b--black-20 tl pb3 pr3">Tâche</th>
              <th className="fw6 bb b--black-20 tl pb3 pr3">Classe</th>
              <th className="fw6 bb b--black-20 tl pb3 pr3">Groupe</th>
              <th className="fw6 bb b--black-20 tl pb3 pr3">Étudiants</th>
            </tr>
          </thead>
          <tbody className="lh-copy">
            {week.tasks.map(task => (
              <tr key={task.task.name}>
                <td className="pv2 pr3 bb b--black-20">{task.task.name.replace(/ /g, '\u00a0')}</td>
                <td className="pv2 pr3 bb b--black-20">{task.classe.replace(/ /g, '\u00a0')}</td>
                <td className="pv2 pr3 bb b--black-20">{task.groupe}</td>
                <td className="pv2 pr3 bb b--black-20">{task.students ? task.students.join(', ') : ''}</td>
              </tr>
                  ))}
          </tbody>
        </table>
      </div>
    );
  }
}

BodyCalendar.defaultProps = {
  week: {},
  selectedWeek: '',
};

BodyCalendar.propTypes = {
  week: PropTypes.object,
  selectedWeek: PropTypes.string,
  fetchWeeksAction: PropTypes.func.isRequired,
};
