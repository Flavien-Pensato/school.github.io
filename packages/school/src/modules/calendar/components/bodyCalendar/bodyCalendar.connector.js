import { connect } from 'react-redux';

import { BodyCalendar } from './bodyCalendar.component';

import { getCurrentWeek } from '../../calendar.selectors';

const mapStateToProps = (state, ownProps) => ({
  week: getCurrentWeek(state, ownProps.selectedWeek),
});

const mapDispatchToProps = dispatch => ({
});

export const BodyCalendarConnected = connect(mapStateToProps, mapDispatchToProps)(BodyCalendar);
