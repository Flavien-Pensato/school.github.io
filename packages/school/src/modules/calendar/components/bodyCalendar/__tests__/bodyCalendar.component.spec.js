import React from 'react';
import { shallow } from 'enzyme';

import { BodyCalendar } from '../bodyCalendar.component';

jest.unmock('../bodyCalendar.component');

describe('Body calendar', () => {
  const props = {};
  const getWrapper = () => shallow(<BodyCalendar {...props} />);

  beforeEach(() => {
    props.fetchWeeksAction = jest.fn();
    props.match = { params: {} };
  });

  it('should render no week found', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('should render normally', () => {
    props.week = {
      tasks: [],
    };
    expect(getWrapper()).toMatchSnapshot();
  });
});
