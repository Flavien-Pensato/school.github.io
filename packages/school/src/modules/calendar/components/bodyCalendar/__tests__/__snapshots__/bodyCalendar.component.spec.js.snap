// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body calendar should render no week found 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BodyCalendar
    fetchWeek={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    match={
      Object {
        "params": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Pas de semaine programmée",
    },
    "ref": null,
    "rendered": "Pas de semaine programmée",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Pas de semaine programmée",
      },
      "ref": null,
      "rendered": "Pas de semaine programmée",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Body calendar should render normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BodyCalendar
    fetchWeek={[MockFunction]}
    match={
      Object {
        "params": Object {},
      }
    }
    week={
      Object {
        "tasks": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        cellSpacing="0"
        className="f7 w-100 center"
      >
        <thead>
          <tr>
            <th
              className="fw6 bb b--black-20 tl pb3 pr3"
            >
              Tâche
            </th>
            <th
              className="fw6 bb b--black-20 tl pb3 pr3"
            >
              Classe
            </th>
            <th
              className="fw6 bb b--black-20 tl pb3 pr3"
            >
              Groupe
            </th>
            <th
              className="fw6 bb b--black-20 tl pb3 pr3"
            >
              Étudiants
            </th>
          </tr>
        </thead>
        <tbody
          className="lh-copy"
        />
      </table>,
      "id": "planning",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "cellSpacing": "0",
        "children": Array [
          <thead>
            <tr>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Tâche
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Classe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Groupe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Étudiants
              </th>
            </tr>
          </thead>,
          <tbody
            className="lh-copy"
          />,
        ],
        "className": "f7 w-100 center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Tâche
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Classe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Groupe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Étudiants
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Tâche
                </th>,
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Classe
                </th>,
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Groupe
                </th>,
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Étudiants
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tâche",
                  "className": "fw6 bb b--black-20 tl pb3 pr3",
                },
                "ref": null,
                "rendered": "Tâche",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Classe",
                  "className": "fw6 bb b--black-20 tl pb3 pr3",
                },
                "ref": null,
                "rendered": "Classe",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Groupe",
                  "className": "fw6 bb b--black-20 tl pb3 pr3",
                },
                "ref": null,
                "rendered": "Groupe",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Étudiants",
                  "className": "fw6 bb b--black-20 tl pb3 pr3",
                },
                "ref": null,
                "rendered": "Étudiants",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "lh-copy",
          },
          "ref": null,
          "rendered": Array [],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          cellSpacing="0"
          className="f7 w-100 center"
        >
          <thead>
            <tr>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Tâche
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Classe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Groupe
              </th>
              <th
                className="fw6 bb b--black-20 tl pb3 pr3"
              >
                Étudiants
              </th>
            </tr>
          </thead>
          <tbody
            className="lh-copy"
          />
        </table>,
        "id": "planning",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cellSpacing": "0",
          "children": Array [
            <thead>
              <tr>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Tâche
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Classe
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Groupe
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Étudiants
                </th>
              </tr>
            </thead>,
            <tbody
              className="lh-copy"
            />,
          ],
          "className": "f7 w-100 center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Tâche
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Classe
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Groupe
                </th>
                <th
                  className="fw6 bb b--black-20 tl pb3 pr3"
                >
                  Étudiants
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    className="fw6 bb b--black-20 tl pb3 pr3"
                  >
                    Tâche
                  </th>,
                  <th
                    className="fw6 bb b--black-20 tl pb3 pr3"
                  >
                    Classe
                  </th>,
                  <th
                    className="fw6 bb b--black-20 tl pb3 pr3"
                  >
                    Groupe
                  </th>,
                  <th
                    className="fw6 bb b--black-20 tl pb3 pr3"
                  >
                    Étudiants
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tâche",
                    "className": "fw6 bb b--black-20 tl pb3 pr3",
                  },
                  "ref": null,
                  "rendered": "Tâche",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Classe",
                    "className": "fw6 bb b--black-20 tl pb3 pr3",
                  },
                  "ref": null,
                  "rendered": "Classe",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Groupe",
                    "className": "fw6 bb b--black-20 tl pb3 pr3",
                  },
                  "ref": null,
                  "rendered": "Groupe",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Étudiants",
                    "className": "fw6 bb b--black-20 tl pb3 pr3",
                  },
                  "ref": null,
                  "rendered": "Étudiants",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "lh-copy",
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
